src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -136,6 +138,26 @@ class LedgerImpl(vm: VM, blockchainConfig: BlockchainConfig) extends Ledger with
         }
   }
 
+  override def simulateTransaction(stx: SignedTransaction, blockHeader: BlockHeader, storages: BlockchainStorages, validators: Validators): TxResult = {
+    val blockchain = BlockchainImpl(storages)
+    val stateRoot = blockchain.getBlockHeaderByHash(blockHeader.hash).map(_.stateRoot)
+    val world = InMemoryWorldStateProxy(storages, stateRoot)
+
+    val gasLimit = stx.tx.gasLimit
+    val config = EvmConfig.forBlock(blockHeader.number, blockchainConfig)
+
+    val context: PC = stx.tx.receivingAddress match {
```

Comment:
The problem was that sender account may not exist (and it's required for the transfer inside `prepareProgramContext`). But I can just make sure there's at least an empty account for sender before calling this fn.